templates/nft-template/move/sources/nft_launchpad.move

```diff
@@ -0,0 +1,506 @@
+module launchpad_addr::nft_launchpad {
+    use std::option;
+    use std::signer;
+    use std::string;
+    use std::vector;
+    use aptos_std::simple_map;
+    use aptos_std::string_utils;
+    use aptos_framework::aptos_account;
+    use aptos_framework::event;
+    use aptos_framework::object;
+    use aptos_token_objects::collection;
+    use aptos_token_objects::royalty;
+    use aptos_token_objects::token;
+
+    use minter::token_components;
+    use minter::mint_stage;
+    use minter::collection_properties;
+    use minter::collection_components;
+
+    /// Sender is not admin
+    const E_NOT_ADMIN: u64 = 1;
+    /// Sender is not pending admin
+    const E_NOT_PENDING_ADMIN: u64 = 2;
+    /// Only admin can update mint fee collector
+    const E_ONLY_ADMIN_CAN_UPDATE_MINT_FEE_COLLECTOR: u64 = 3;
+    /// Only admin can create FA
+    const E_ONLY_ADMIN_CAN_CREATE_FA: u64 = 4;
+    /// No mint limit
+    const E_NO_MINT_LIMIT: u64 = 5;
+    /// Mint limit reached
+    const E_MINT_LIMIT_REACHED: u64 = 6;
+    /// No active mint stages
+    const E_NO_ACTIVE_STAGES: u64 = 7;
+
+    const ALLOWLIST_MINT_STAGE_CATEGORY: vector<u8> = b"Allowlist mint stage";
```

Comment:
>I'm also worried that if you change the literal here, it will break existing NFT launches that are trying to look up the stage in the simple map
This contract is meant to be instantiated for each launchpad, not to be used as an sdk, so I don't think updating it will break existing collections. For deployed contract, changing value of constants is incompatible.

Reply:
It's not true, we can change constants during upgrade

Reply:
ah i probably changed something else that triggered the incompatible error

Reply:
Ideally enums would be good to use here. I don't have a strong preference here between using Strings vs numbers for differentiation, and think devex is somewhat similar. If you think using numbers vs Strings is better devex, let me know!